services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,261 @@
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.meter.SwitchMeterEntries
+import org.openkilda.testing.Constants
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.northbound.NorthboundService
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Ignore
+import spock.lang.Issue
+import spock.lang.Narrative
+import spock.lang.Shared
+import spock.lang.Unroll
+
+@Narrative("The test suite checks if traffic meters, including default, are set and deleted in a correct way.")
+class MetersSpec extends BaseSpecification {
+
+    static DISCO_PKT_RATE = 200 // Number of packets per second for the default flows
+    static DISCO_PKT_SIZE = 250 // Default size of the discovery packet
+    static NOVI_BURST_MIN = 1.004 // Guaranteed minimum flow burst rate on a Noviflow switch, with rounding errors.
+    static NOVI_BURST_MAX = 1.051 // Possible maximum flow burst rate on a Noviflow switch, with rounding errors.
+    static DISCO_PKT_BURST = 4096 // Default desired packet burst rate for the default flows (ignored by Noviflow)
+
+    @Autowired
+    TopologyDefinition topology
+
+    @Autowired
+    NorthboundService northboundService
+
+    @Shared
+    def centecs
+
+    @Shared
+    def nonCentecs
+
+    def setupOnce() {
+        this.requireProfiles("hardware") //TODO: remove as soon as OVS 2.10 + kernel 4.18+ get wired in
+    }
+
+    List<Switch> getNonCentecSwitches() {
+        if (!nonCentecs) {
+            def nonCentecSw = northboundService.getActiveSwitches().findAll {
+                !it.description.toLowerCase().contains("centec")
+            }*.switchId
+            nonCentecs = topology.getActiveSwitches().findAll {
+                (it.ofVersion == "OF_13") && (nonCentecSw.contains(it.dpId))
+            }
+        }
+        return nonCentecs
+    }
+
+    List<Switch> getCentecSwitches() {
+        if (!centecs) {
+            centecs = topology.getActiveSwitches().findAll { it.ofVersion == "OF_13" } - getNonCentecSwitches()
+        }
+        return centecs
+    }
+
+    @Unroll
+    def "Unable to delete meter with invalid ID #meterId on a #switchType switch"() {
+        when: "Try to delete meter with invalid ID"
+        northboundService.deleteMeter(sw.getDpId(), -1)
+
+        then: "Got BadRequest because meter ID is invalid"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        where:
+        meterId | sw                        | switchType
+        -1      | getNonCentecSwitches()[0] | "non-Centec"
+        0       | getCentecSwitches()[0]    | "Centec"
+        -1      | getCentecSwitches()[0]    | "non-Centec"
+        0       | getNonCentecSwitches()[0] | "Centec"
```

Comment:
I don't think that this is switch-dependent. But you can leave it if you want since it's relatively fast